src/strategy.py

```diff
@@ -34,7 +34,7 @@ def __init__(self, constants, start_time, **kwargs):
         # relative allowed difference between battery SoC and desired SoC when leaving
         self.margin = 0.05
         self.allow_negative_soc = False
-        self.DISCHARGE_LIMIT = 0
+        self.DISCHARGE_LIMIT = kwargs.get('discharge_limit')  # default always exists
```

Comment:
Makes sense. I used your suggestions as it allows for DISCHARGE LIMIT to be set in strategy options providing more flexibility for the user and providing DISCHARGE_LIMIT before running simulation.py is only relevant for the schedule strategy. 
In my original PR, I thought it might be cleaner if DISCHARGE_LIMIT could be set in one place only to avoid confusion, in case for example a user sets DISCHARGE_LIMIT in generate.cfg and accidentally overrides it later with a STRAT_OPT. Should we emit a warning if DISCHARGE_LIMIT is set in generate.cfg and then overridden by a STRAT_OPTION?

Reply:
Providing this flexibility sounds a good. In my opinion a warning should be emitted, simply to avoid confusion or mistakes.